### ISSUE TITLE
No straightforward way to set the right locale for MO files not named after it when joining

### ISSUE THREAD

**Issue by ximion** [itstool/itstool#44]
Hi!
When using Meson (but also quite a few other tools), MO files are generated from their PO files into a directory structure like `po/<locale>/LC_MESSAGES/<domain>.mo`. Due to that, joining XML files with these MO files does not work without renaming them after their respective locale first, usually in a temporary directory.

Unfortunately, Itstool can not join from PO files, which would not create this issue (as those files have the locale information embedded).
Also, you can not specify the `--lang=LANGUAGE` option multiple times / once per file when joining multiple locale at once, so you would either generate a lot of temporary XML files and run Itstool on each one at at time (extremely slow) or use the rename trick.

Can this workflow be improved somehow? I could imagine that guessing the locale from the full MO path may be an option if `LC_MESSAGES` is present, alternatively some magic to specify the locale per file on the command-line may also work.
The ability to join XML from PO files instead of MO files may also fix this and may also be quite nice to have.

Thank you for considering this feature request!

**Comment by shaunix** [comment]
I think this would be addressed by #32 . Basically, the language information is in the MO file, sort of, if you assume metadata is embedded the way gettext embeds it. It's pretty easy to extract it, and I just never got around to putting that line of code in.

Also, yeah, it would be pretty great to be able to go directly from PO files and just compile them to MO automatically in the background.

**Comment by ximion** [comment]
Makes sense to me :-) I can't look into this at the moment and propose a patch, but maybe in future.
Thank you for creating itstool, it is extremely useful!
